packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js

```diff
@@ -1,23 +1,157 @@
-import React from 'react';
+import React, { useCallback } from 'react';
+import { Form } from 'informed';
+import { func, shape, string } from 'prop-types';
+import { useToasts } from '@magento/peregrine';
+import { useCreateAccount } from '@magento/peregrine/lib/talons/CheckoutPage/OrderConfirmationPage/useCreateAccount';
 
-import classes from './orderConfirmationPage.css';
+import combine from '../../../util/combineValidators';
+import { mergeClasses } from '../../../classify';
+import {
+    hasLengthAtLeast,
+    isRequired,
+    validatePassword
+} from '../../../util/formValidators';
+
+import Button from '../../Button';
+import Checkbox from '../../Checkbox';
+import Field from '../../Field';
+import TextInput from '../../TextInput';
+import defaultClasses from './createAccount.css';
+
+import CREATE_ACCOUNT_MUTATION from '../../../queries/createAccount.graphql';
+import CREATE_CART_MUTATION from '../../../queries/createCart.graphql';
+import GET_CART_DETAILS_QUERY from '../../../queries/getCartDetails.graphql';
+import GET_CUSTOMER_QUERY from '../../../queries/getCustomer.graphql';
+import SIGN_IN_MUTATION from '../../../queries/signIn.graphql';
+
+const CreateAccount = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const [, { addToast }] = useToasts();
+
+    const onSubmit = useCallback(() => {
+        // TODO: Redirect to account/order page when implemented.
+        window.scrollTo({
+            left: 0,
+            top: 0,
+            behavior: 'smooth'
+        });
+
+        addToast({
+            type: 'info',
+            message: 'Account successfully created.',
+            timeout: 5000
+        });
+    }, [addToast]);
+
+    const talonProps = useCreateAccount({
+        queries: {
+            createAccountQuery: CREATE_ACCOUNT_MUTATION,
+            customerQuery: GET_CUSTOMER_QUERY
+        },
+        mutations: {
+            createCartMutation: CREATE_CART_MUTATION,
+            getCartDetailsQuery: GET_CART_DETAILS_QUERY,
+            signInMutation: SIGN_IN_MUTATION
+        },
+        initialValues: {
+            email: props.email,
+            firstName: props.firstname,
+            lastName: props.lastname
+        },
+        onSubmit
+    });
+
+    const { errors, handleSubmit, isDisabled, initialValues } = talonProps;
+
+    // Map over any errors we get and display an appropriate error.
+    const errorMessage = errors.length
+        ? errors
+              .map(({ message }) => message)
+              .reduce((acc, msg) => msg + '\n' + acc, '')
```

Comment:
Consider using [join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) for this.

Reply:
Done in ff792f8